bigquery/tests/unit/test_client.py

```diff
@@ -5161,3 +5161,84 @@ def test__do_multipart_upload_wrong_size(self):
 
         with pytest.raises(ValueError):
             client._do_multipart_upload(file_obj, {}, file_obj_len + 1, None)
+
+    def test__schema_from_json(self):
+        from google.cloud.bigquery.schema import SchemaField
+
+        file_content = """[
+          {
+            "description": "quarter",
+            "mode": "REQUIRED",
+            "name": "qtr",
+            "type": "STRING"
+          },
+          {
+            "description": "sales representative",
+            "mode": "NULLABLE",
+            "name": "rep",
+            "type": "STRING"
+          },
+          {
+            "description": "total sales",
+            "mode": "NULLABLE",
+            "name": "sales",
+            "type": "FLOAT"
+          }
+        ]"""
+        expected = list()
+        json_data = json.loads(file_content)
+
+        for field in json_data:
+            schema_field = SchemaField.from_api_repr(field)
+            expected.append(schema_field)
+
+        client = self._make_client()
+        mock_file_path = "/mocked/file.json"
```

Comment:
In addition to a test that passes in a path, please add a test `test_schema_from_json_with_file_object` that passes in a file object. That should actually be easier to test because you can pass in a [`BytesIO`](https://pythonhosted.org/six/#six.BytesIO) object. No mocking required for that second test.

Same applies for `schema_to_json`. Add a test using `BytesIO` for that, too, please.

Reply:
Added two tests to cover this but i had to use StringIO to make it work.